### PR TITLE
fix: remove current page link option in page reference

### LINKED ISSUE #3524: [FR] Add a vertical scrollbar for page link menu and hide the current page
### Description

> When I entered @ for a link to a page, I was confused because the window only showed five pages. I then realized that I had to enter more letters. Then the desired page was also displayed 😃 . Would it be useful to have a vertical scrollbar in the window to show that there are more pages to choose from?  And another hint: In the window also the page is shown where I insert the link. This page should not be available for selection. Keep up the good work 👍 ! 
by MacMeikel, Copied from Discord.

image

Tasks:
- [ ] Add vertical scrollbar to page link menu 
- [ ] Remove the current page as an option from the menu

### Impact

Less confusion while using Page Link Menu

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by narayann7**
### Feature Preview

image

fixes #3524

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by narayann7** [comment]
@Xazin can you please review the code changes for this fix

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart [hunk 1]
```diff
@@ -57,7 +57,7 @@ class _AppFlowyEditorPageState extends State<AppFlowyEditorPage> {
   late final InlineActionsService inlineActionsService = InlineActionsService(
     context: context,
     handlers: [
-      InlinePageReferenceService().inlinePageReferenceDelegate,
+      InlinePageReferenceService(context).inlinePageReferenceDelegate,
```
[the remark this answers is not in the document]
Comment (narayann7) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart]: yeah make sense, changed and pushed the code.

**Comment by narayann7** [comment]
@Xazin can you check why these test are failing, there were no much changes in code

**Comment by narayann7** [comment]
i think due to frame issue test is failing, re running will clear the test

**Review by Xazin — APPROVED** [review]

**Pull request merged**